OsmAnd/src/net/osmand/plus/mapcontextmenu/MenuBuilder.java

```diff
@@ -103,6 +99,74 @@
 	private String preferredMapLang;
 	private String preferredMapAppLang;
 	private boolean transliterateNames;
+	private static final int PICK_IMAGE = 1231;
+	private static final Log LOG = PlatformUtil.getLog(MenuBuilder.class);
+	private View view;
+
+	private GetImageCardsListener imageCardListener = new GetImageCardsListener() {
+		private String[] idOfCurrentPlace = new String[0];
+
+		@Override
+		public void onPostProcess(List<ImageCard> cardList) {
+			processOnlinePhotosCards(cardList);
+		}
+
+		@Override
+		public void onOPRPlaceIdAcquired(String[] id) {
+			idOfCurrentPlace = id;
+			firstRow = false;
+			view.post(new Runnable() {
+				@Override
+				public void run() {
+					buildUploadImagesRow(view);
+				}
+			});
+		}
+
+		@Override
+		public void uploadImageToPlace(View view, InputStream image) {
+			String url = "https://test.openplacereviews.org/api/ipfs/image";
+			String response = NetworkUtils.sendPostDataRequest(url, image);
+			if (response != null) {
+				int res = 0;
+				try {
+					res = SecUtils.uploadImage(
```

Comment:
there is no such method in SecUtils upload image, so it should not be added.

Don't modify copy/pasted classes! Also copy/pasted classes should have header at which revision there were copied.

Reply:
1) It is just renamed version of main()
2) it is impossible not to modify the classes because of the difference between pure java and android platforms